src/PhpSessionPersistence.php

```diff
@@ -89,6 +93,63 @@ public function __construct(bool $nonLocking = false, bool $deleteCookieOnEmptyS
         $this->cookieSameSite = ini_get('session.cookie_samesite');
     }
 
+    /**
+     * @param array $session
```

Comment:
This parameter could be improved with something like:

```suggestion
     * @param array{
     *   cache_limiter?: string,
     *   other-keys?: whatever,
     *   persistence?: array{
     *     ext?: array{
     *       delete_cookie_on_empty_session?: bool,
     *       other-keys?: whatever,
     *     },
     *   },
     * } $session
```

If declared as a `/** @psalm-type SessionConfig = array{...} */` at class level, the array shape can be imported easily anywhere it's needed - example in [laminas-input-filter](https://github.com/laminas/laminas-inputfilter/blob/a79c31bcb58d1041393dd3a7a8187829e4ca7b02/src/InputFilterInterface.php#L14)

Reply:
I created psalm type `SessionConfig`, but I could not make it work correctly in `src/PhpSessionPersistenceFactory.php` so I had to psalm suppress `MixedArgumentTypeCoercion` there.

Reply:
It's acceptable to suppress the issue in the factory. Alternatives are validate the array shape in code somewhere, force the type with `/** @psalm-var SessionConfig */`, or expand the baseline - generally speaking, expanding the baseline is preferred in cases like this 👍